projects/snapshot_bot/snapshot_bot.cc

```diff
@@ -259,16 +261,18 @@ class RobotFSM : FSM<State>::StateHandler
                     }
 
                     const double elapsed = static_cast<millisecond_t>(m_RecordingStopwatch.elapsed()).value();
-
                     // If Vicon tracking is available
                     if(m_Config.shouldUseViconTracking()) {
 #ifdef USE_VICON
                         // Get tracking data
                         const auto objectData = m_ViconTracking.getObjectData(m_Config.getViconTrackingObjectName());
+                        const auto imuData = imu.getEulerAngles();
                         m_Recorder->record(objectData.getPose(),
                                            m_Output,
                                            elapsed,
-                                           objectData.getFrameNumber());
+                                           objectData.getFrameNumber(),
+                                           imuData);
```

Comment:
This is a bug. The reason this compiles is because the ``USE_VICON`` option is not enabled, so the compiler never actually runs into this code. What you need to do is build like so:
```
cmake -DUSE_VICON=ON ..
# ... etc.
```

The problem is that you're passing a ``std::array<degree_t, 3>`` and the compiler needs to be able to convert this to a string with ``std::to_string``, which it won't be able to. This should be:
```c++
imuData[0].value(), imuData[1].value(), imuData[2].value()
```

(Don't forget the ``.value()`` or you'll end up with something like ``1 deg, 2 deg, 0.5 deg`` written into your CSV file.)